File: webhooks/serve_webhook_agent.py
#!/usr/bin/env python3
"""Serve the local database tools used by the post-webhook agent."""

from conductor.ai.agents import AgentRuntime
from conductor.client.configuration.configuration import Configuration

from .utils.webhook_agent import webhook_agent


def main() -> None:
    """Deploy the agent and serve its tool workers until interrupted."""
    with AgentRuntime(configuration=Configuration()) as runtime:
        runtime.serve(webhook_agent)


if __name__ == "__main__":
    main()

File: webhooks/typescript/serveWebhookAgent.ts
/** Serve the local database tools used by the post-webhook agent. */

import { AgentRuntime } from "@io-orkes/conductor-javascript/agents";
import { webhookAgent } from "./utils/webhookAgent.ts";

/** Deploy the agent and serve its tool workers until interrupted. */
async function main(): Promise<void> {
  const runtime = new AgentRuntime();
  try {
    console.log("Serving the agent's tool workers. Press Ctrl+C to stop.");
    // Resolves once Ctrl+C has stopped the workers.
    await runtime.serve(webhookAgent);
  } finally {
    // Also stops the background token refresh of the client the runtime created, which would
    // otherwise keep this process running.
    await runtime.client.close();
  }
}

await main();
